### PR TITLE
Add migration support for name mapping, type mapping, relation and generated annotations

### PR DESCRIPTION
**Pull request by hasathcharu**
## Purpose
This is the second PR for the issue mentioned below. This adds support for the following annotations when using migrate command.
- `@sql:Name`
- `@sql:Char`
- `@sql:Varchar`
- `@sql:Decimal`
- `@sql:Relation`
- `@sql:Generated`

Fixes: [/ballerina-platform/ballerina-library/issues/6189](https://github.com/ballerina-platform/ballerina-library/issues/6189)

## Checklist
- [X] Linked to an issue
- [ ] Updated the specification
- [X] Updated the changelog
- [X] Added tests

### CODE REVIEW

File: persist-cli/src/main/java/io/ballerina/persist/models/Relation.java [hunk 1]
```diff
@@ -170,5 +171,29 @@ public String getColumnName() {
...
+            return Objects.equals(field, key.field) && Objects.equals(columnName, key.columnName) &&
+                    Objects.equals(reference, key.reference) &&
+                    Objects.equals(referenceColumnName, key.referenceColumnName) && Objects.equals(type, key.type);
+        }
+        @Override
+        public int hashCode() {
+            return Objects.hash(field, columnName, reference, referenceColumnName, type);
+        }
+
+        public boolean onlyColumnNameChanged(Key key) {
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/models/Relation.java:193]:
```suggestion
        public boolean isOnlyColumnRenamed(Key key) {
```

File: persist-cli/src/main/java/io/ballerina/persist/models/ModelDifferences.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Class to process and store migrate differences.
+ *
+ *
+ */
+public class ModelDifferences {
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/models/ModelDifferences.java:32]:
```suggestion
public class MigrationProcessor {
```

Comment (daneshk) [reply persist-cli/src/main/java/io/ballerina/persist/models/ModelDifferences.java:32]: how about changing the class name to either `MigrationProcessor` or `MigrationManager`

File: persist-cli/src/main/java/io/ballerina/persist/cmd/Migrate.java [hunk 3]
```diff
@@ -515,49 +515,71 @@ private static List<String> findDifferences(Module previousModel, Module current
...
+                currentModel, queries);
+        addCreateForeignKeyQueries(modelDifferences.getAddedForeignKeys(), queries);
+        addModifyColumnTypeQueries(modelDifferences.getChangedFieldTypes(), queries);
+
+        printDetailedListOfDifferences(modelDifferences.getDifferences());
+
+        return queries;
+    }
+
+    private static boolean onlyColumnNamesChanged(List<Relation.Key> previousKeys, List<Relation.Key> currentKeys) {
```
Comment (daneshk) [inline persist-cli/src/main/java/io/ballerina/persist/cmd/Migrate.java:543]:
```suggestion
    private static boolean isOnlyColumnRenamed(List<Relation.Key> previousKeys, List<Relation.Key> currentKeys) {
```

File: persist-cli/src/main/java/io/ballerina/persist/models/ModelDifferences.java (hunk 2 above)
Comment (hasathcharu) [reply persist-cli/src/main/java/io/ballerina/persist/models/ModelDifferences.java:32]: In that case is it okay to keep it in the model package?

Comment (daneshk) [reply persist-cli/src/main/java/io/ballerina/persist/models/ModelDifferences.java:32]: Let's move it to utils package.

Comment (hasathcharu) [reply persist-cli/src/main/java/io/ballerina/persist/models/ModelDifferences.java:32]: This was agreed to keep in the `models` package as it only acts as a DataHolder for `migrate` and renamed as `MigrationDataHolder`.

**Review by daneshk — APPROVED** [review]

**Pull request merged**